### PR TITLE
Implement fullwidth styling to FileSelector and FileDownload

### LINKED ISSUE #474: No possiblity to change width of file_selector and file_upload
## Issue

There is no possibility to extend the file_selector or file_upload to the full width of the page.

## How to replicate

image

```python
from taipy.gui import Gui


page = """
<|{None}|file_selector|label=Selector|width=100%|class_name=fullwidth|>

<|{None}|file_download|label=Download|width=100%|class_name=fullwidth|>
"""

if __name__ == "__main__":
    Gui(page=page).run()
```

## Expected behavior

This should take all the width of the page.

## Environment
Taipy 3.0.0

### PR DESCRIPTION
**Pull request by Satoshi-Sh**
## Related Issue 
Resolves #474

##  Updates 
- Added styling files 
- Moved className to the Button component from the Label component.

## Screenshot 
Screenshot from 2024-06-03 08-29-23

### CODE REVIEW

File: frontend/taipy-gui/src/components/Taipy/FileDownload.tsx [hunk 1]
```diff
@@ -93,11 +93,11 @@ const FileDownload = (props: FileDownloadProps) => {
     const aProps = useMemo(() => (bypassPreview ? {} : { target: "_blank", rel: "noreferrer" }), [bypassPreview]);
 
     return render ? (
-        <label htmlFor={linkId} className={className}>
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/FileDownload.tsx]: I'm afraid that moving `className` to the button will cause breaking change for the current users of Taipy.
The point of having the `className` at the top-level of the component tree of a Taipy element is to facilitate the style customization of any of the component in the element's tree.

File: frontend/taipy-gui/public/stylekit/controls/file_download.css [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                   TAIPY FILE_DOWNLOAD
+
+***************************************************************/
+
+/*************************************************
+              MODIFIER CLASSES
+**************************************************/
+
+/* fullwidth :  */
+.taipy-file-download.fullwidth {
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/public/stylekit/controls/file_download.css]: I propose to keep the `className` at the same level and define the style as `.taipy-file-download.fullwidth button {`

File: frontend/taipy-gui/public/stylekit/controls/file_selector.css [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+                   TAIPY FILE_SELECTOR
+
+***************************************************************/
+
+/*************************************************
+              MODIFIER CLASSES
+**************************************************/
+
+/* fullwidth :  */
+.taipy-file-selector.fullwidth {
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/public/stylekit/controls/file_selector.css]: idem

File: frontend/taipy-gui/src/components/Taipy/FileSelector.tsx [hunk 4]
```diff
@@ -144,7 +144,7 @@ const FileSelector = (props: FileSelectorProps) => {
     }, [handleDrop, handleDragLeave, handleDragOverWithLabel]);
 
     return (
-        <label htmlFor={inputId} className={className}>
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/FileSelector.tsx]: idem

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
I very much like the way you approach this.
I think that we cannot take the risk of breaking existing Taipy apps.

**Review by FredLL-Avaiga — APPROVED** [review]
Excellent tx @Satoshi-Sh

**Review by FabienLelaquais — APPROVED** [review]
Neat!

**Comment by FredLL-Avaiga** [comment]
@Satoshi-Sh , can I merge this ?

**Comment by Satoshi-Sh** [comment]
> I'm afraid that moving className to the button will cause breaking change for the current users of Taipy.
The point of having the className at the top-level of the component tree of a Taipy element is to facilitate the style customization of any of the component in the element's tree.

Your explanation makes sense, but I have a question @FredLL-Avaiga. 

If we have the class name at the top level, we cannot style fileSelector and fileDownload by CSS as ClassName is in the label. Is it fine?

**Comment by FredLL-Avaiga** [comment]
> > I'm afraid that moving className to the button will cause breaking change for the current users of Taipy.
> > The point of having the className at the top-level of the component tree of a Taipy element is to facilitate the style customization of any of the component in the element's tree.
> 
> Your explanation makes sense, but I have a question @FredLL-Avaiga.
> 
> If we have the class name at the top level, we cannot style fileSelector and fileDownload by CSS as ClassName is in the label. Is it fine?

I don't understand your point.
the className is associated to the label that is the top of the components' tree so we can access everything from there
like `.taipy-file-download button`

**Comment by Satoshi-Sh** [comment]
Thanks. I got it. I will update another branch soon. @FredLL-Avaiga

**Comment by Satoshi-Sh** [comment]
Great. Thanks for the guidance. 

Do we need to add a similar description to the file_selector and file_download doc page?
Screenshot from 2024-06-05 06-07-53

**Comment by FredLL-Avaiga** [comment]
I guess you can create a taipy-doc issue

**Pull request merged**